net/respondd/src/respondd.c

```diff
@@ -320,27 +390,92 @@ static struct json_object * handle_request(char *request, bool *compress) {
 	}
 }
 
-
-static void serve(int sock) {
-	char input[256];
-	const char *output = NULL;
-	ssize_t input_bytes, output_bytes;
+// the return value indicates whether there was a request
+static bool accept_request(struct request_schedule *schedule, int sock,
+                           uint64_t timeout, uint64_t max_multicast_delay) {
```

Comment:
I think it makes more sense to calculate the timeout inside this function, as it is computable from the schedule which this function has to have access to anyway.